### PR TITLE
matis/amam-63-je-souhaite-pouvoir-me-connecter-et-agir-en-tant

### PR DESCRIPTION
**Pull request by matisbyar**

### CODE REVIEW

**Comment by matisbyar** [comment]
Maybe should I explain this PR a bit as I heard it does not appear clear enough.

There's an issue with the Supabase own SSR middleware that is causing it to be **unable to interact with the user's session**. As a result, even if connected to any OAuth provider successfully one user would not be authenticated in the app.

To avoid this issue, I had to implement my **own way of securing routes**. Therefore, in the App Router, lies two paths `(protected)` and `(unprotected)`. Any route in the (protected) directory **requires auth**; this is handled by the `layout.tsx` present at the root of `(protected)`.

Here is the security:
```tsx
// src/app/(protected)/layout.tsx

// ...
const {error} = await createClient().auth.getUser();

if (error) {
    redirect("/login");
}

// ...
```

I could have handled this without creating said protected and unprotected directories, but without that implementation, there would not be any way of accessing the login route without being redirected endlessly.

The unprotected directory could be used to handle various public pages such as legal stuff.

Hope that makes it more clear.

File: src/app/(unprotected)/login/actions.ts
Comment (maxbodin) [inline src/app/(unprotected)/login/actions.ts:1]: Maybe refactoring this two functions in two separated files would be clearer ? login.ts and signOut.ts ?

Comment (matisbyar) [reply src/app/(unprotected)/login/actions.ts:1]: I don't think dissociating two related actions in their own files would be a great approach; especially knowing they are located in the login directory.

This refactoring proposition does not affect code readability nor is subject to change—lmk what you think.

File: src/app/api/auth/callback/route.ts
[the remark this answers is not in the document]
Comment (matisbyar) [reply src/app/api/auth/callback/route.ts:1]: J'ai copié-collé le tutoriel de Supabase, ils seront ravis de lire ta PR pour changer leur doc. Peut-être mettront-ils moins d'une semaine à la lire.

[the remark this answers is not in the document]
Comment (matisbyar) [reply src/app/api/auth/callback/route.ts:1]: De plus, si tu lisais mes autres commentaires, tu contestera que c'est le cas.

File: src/app/(unprotected)/login/actions.ts
Comment (maxbodin) [reply src/app/(unprotected)/login/actions.ts:1]: I see where you're coming from, but separating the two actions into their own files actually helps with clarity and maintainability. Even though they're in the same login directory, keeping related but distinct actions in their own files makes the purpose of each file more obvious. This way, it’s easier for someone new to the codebase (or even future you) to quickly find and understand what each file does.

By contrast, putting multiple actions into one file with a less clear name (actions) could make it harder to locate specific functionality or understand the scope of changes. If we ever need to modify or extend one action, it’s easier to do so when each function has its own dedicated space, which keeps the code organized and scalable.

Let me know your thoughts!

File: src/app/api/auth/callback/route.ts
[the remark this answers is not in the document]
Comment (maxbodin) [reply src/app/api/auth/callback/route.ts:1]: Ah, je comprends mieux maintenant ! C’est vrai que si c’est un simple copier-coller du tutoriel de Supabase, ça change un peu les choses. Cela dit, indépendamment de l’origine du texte, l’idée serait juste de s’assurer que les commentaires suivent une convention commune dans notre code, histoire de maintenir une certaine cohérence, surtout si on doit y revenir plus tard. 

Et effectivement, j’ai bien vu que tu fais attention dans tes autres commentaires, c’était simplement une petite remarque pour aligner tout ça. 😊

File: src/app/layout.tsx
Comment (maxbodin) [inline src/app/layout.tsx:1]: Pourquoi avoir retiré les lignes suivantes ? 

export const metadata: Metadata = {
    title: "CinéFIL",
    description: "Site de votre cinéma CinéFIL à IMT Atlantique",
    icons: [{rel: "icon", url: "/favicon.ico"}],

File: src/app/api/auth/callback/route.ts
[the remark this answers is not in the document]
Comment (matisbyar) [reply src/app/api/auth/callback/route.ts:1]: On en parle de tes push qui ne build pas parce que tu as désactivé ton ESLint ? Où sont les conventions communes maintenant ? 😄

File: src/app/layout.tsx
Comment (matisbyar) [reply src/app/layout.tsx:1]: Je les ai remplacées pour les "aligner" à la DA de cinéfil.

De plus, Next gère automatiquement le favicon quand il le trouve.

File: src/app/(unprotected)/login/actions.ts
Comment (maxbodin) [reply src/app/(unprotected)/login/actions.ts:1]: @matisbyar Hi, it seems like you have edited my comment. Could you let me know if there was a reason for the changes? I'm happy to discuss further if needed!

File: src/app/api/auth/callback/route.ts
[the remark this answers is not in the document]
Comment (maxbodin) [reply src/app/api/auth/callback/route.ts:1]: Ah, tu marques un point ! 😅 C’est vrai que ce push n’était pas parfait, j’aurais dû faire plus attention à l'ESLint. C’était un cas isolé, mais tu as raison, on doit tous respecter les conventions pour éviter ce genre de soucis.

File: src/app/layout.tsx
Comment (maxbodin) [reply src/app/layout.tsx:1]: Je comprends mieux maintenant, mais il me semble qu'on n'a pas vraiment décidé de cette DA ensemble. J'aurais préféré qu'on en discute avant de faire ces changements, surtout pour être alignés dès le départ. Concernant le favicon, je n'étais pas au courant que Next le gérait automatiquement—merci pour l'info !

**Pull request merged**